include/revng/Support/IRHelpers.h

```diff
@@ -1575,3 +1576,19 @@ std::unique_ptr<llvm::Module> cloneIntoContext(const llvm::Module &Module,
 /// modules being merged.
 void linkFunctionModules(std::unique_ptr<llvm::Module> &&Source,
                          std::unique_ptr<llvm::Module> &Destination);
+
+/// Helper class that allows getting the exhaustive list of functions called by
+/// a function
+class FunctionCallEnumerator {
+private:
+  const llvm::Module &Module;
+  std::map<const llvm::Function *, std::set<const llvm::Function *>>
+    CalledFunctions;
+
+public:
+  FunctionCallEnumerator(const llvm::Module &Module) : Module(Module) {}
+
+  RecursiveCoroutine<std::set<const llvm::Function *>>
```

Comment:
1. Return a `const &`, not a copy.
2. Do not expose `RecursiveCoroutine` in `IRHelpers.h`.